===================================================================
--- src/v2EthLendingPoolCollateralManager/LendingPoolCollateralManager/contracts/protocol/libraries/logic/ValidationLogic.sol
+++ src/v2AvaLendingPoolCollateralManager/LendingPoolCollateralManager/contracts/protocol/libraries/logic/ValidationLogic.sol
@@ -96,22 +96,15 @@
         );
     }
 
     struct ValidateBorrowLocalVars {
-        uint256 principalBorrowBalance;
         uint256 currentLtv;
         uint256 currentLiquidationThreshold;
-        uint256 requestedBorrowAmountETH;
         uint256 amountOfCollateralNeededETH;
         uint256 userCollateralBalanceETH;
         uint256 userBorrowBalanceETH;
-        uint256 borrowBalanceIncrease;
-        uint256 currentReserveStableRate;
         uint256 availableLiquidity;
-        uint256 finalUserBorrowRate;
         uint256 healthFactor;
-        DataTypes.InterestRateMode rateMode;
-        bool healthFactorBelowThreshold;
         bool isActive;
         bool isFrozen;
         bool borrowingEnabled;
         bool stableRateBorrowingEnabled;
@@ -212,9 +205,9 @@
          *    they are borrowing, to prevent abuses.
          * 3. Users will be able to borrow only a portion of the total available liquidity
          **/
 
-        if (vars.rateMode == DataTypes.InterestRateMode.STABLE) {
+        if (interestRateMode == uint256(DataTypes.InterestRateMode.STABLE)) {
             //check if the borrow mode is stable and if stable rate borrowing is enabled on this reserve
 
             require(
                 vars.stableRateBorrowingEnabled,
