diff --git a/tmp/1_0x368e6441bB27159c6e8e6d3bbd9147BEcBA915E3_flat/LendingPoolCollateralManager.sol b/tmp/1_0xcc9632725266473ab5d42320E8Ae00437A1df3bB_flat/LendingPoolCollateralManager.sol
index f47e37a..d9ea585 100644
--- a/tmp/1_0x368e6441bB27159c6e8e6d3bbd9147BEcBA915E3_flat/LendingPoolCollateralManager.sol
+++ b/tmp/1_0xcc9632725266473ab5d42320E8Ae00437A1df3bB_flat/LendingPoolCollateralManager.sol
@@ -36,7 +36,6 @@ contract LendingPoolCollateralManager is ILendingPoolCollateralManager, Versione
     using WadRayMath for uint256;
     using PercentageMath for uint256;
 
-    uint256 internal constant LIQUIDATION_CLOSE_FACTOR_PERCENT = 5000;
     ILiquidationsGraceSentinel public immutable override LIQUIDATIONS_GRACE_SENTINEL;
 
     struct LiquidationCallLocalVars {
@@ -125,8 +124,7 @@ contract LendingPoolCollateralManager is ILendingPoolCollateralManager, Versione
 
         vars.userCollateralBalance = vars.collateralAtoken.balanceOf(user);
 
-        vars.maxLiquidatableDebt =
-            vars.userStableDebt.add(vars.userVariableDebt).percentMul(LIQUIDATION_CLOSE_FACTOR_PERCENT);
+        vars.maxLiquidatableDebt = vars.userStableDebt.add(vars.userVariableDebt);
 
         vars.actualDebtToLiquidate = debtToCover > vars.maxLiquidatableDebt ? vars.maxLiquidatableDebt : debtToCover;
 
