===================================================================
--- src/v2EthLendingPoolCollateralManager/LendingPoolCollateralManager/contracts/interfaces/IReserveInterestRateStrategy.sol
+++ src/v2PolLendingPoolCollateralManager/LendingPoolCollateralManager/contracts/interfaces/IReserveInterestRateStrategy.sol
@@ -12,13 +12,24 @@
     function getMaxVariableBorrowRate() external view returns (uint256);
 
     function calculateInterestRates(
         address reserve,
-        uint256 utilizationRate,
+        uint256 availableLiquidity,
         uint256 totalStableDebt,
         uint256 totalVariableDebt,
         uint256 averageStableBorrowRate,
         uint256 reserveFactor
+    ) external view returns (uint256, uint256, uint256);
+
+    function calculateInterestRates(
+        address reserve,
+        address aToken,
+        uint256 liquidityAdded,
+        uint256 liquidityTaken,
+        uint256 totalStableDebt,
+        uint256 totalVariableDebt,
+        uint256 averageStableBorrowRate,
+        uint256 reserveFactor
     )
         external
         view
         returns (
