===================================================================
--- src/v2EthLendingPoolCollateralManager/LendingPoolCollateralManager/contracts/protocol/libraries/logic/ReserveLogic.sol
+++ src/v2PolLendingPoolCollateralManager/LendingPoolCollateralManager/contracts/protocol/libraries/logic/ReserveLogic.sol
@@ -224,20 +224,18 @@
         vars.totalVariableDebt = IVariableDebtToken(
             reserve.variableDebtTokenAddress
         ).scaledTotalSupply().rayMul(reserve.variableBorrowIndex);
 
-        vars.availableLiquidity = IERC20(reserveAddress).balanceOf(
-            aTokenAddress
-        );
-
         (
             vars.newLiquidityRate,
             vars.newStableRate,
             vars.newVariableRate
         ) = IReserveInterestRateStrategy(reserve.interestRateStrategyAddress)
             .calculateInterestRates(
                 reserveAddress,
-                vars.availableLiquidity.add(liquidityAdded).sub(liquidityTaken),
+                aTokenAddress,
+                liquidityAdded,
+                liquidityTaken,
                 vars.totalStableDebt,
                 vars.totalVariableDebt,
                 vars.avgStableRate,
                 reserve.configuration.getReserveFactor()
