===================================================================
--- src/contracts/interfaces/ILendingPool.sol
+++ contracts/v2AvaPool/LendingPool/contracts/interfaces/ILendingPool.sol
@@ -3,9 +3,9 @@
 pragma experimental ABIEncoderV2;
 
 import {ILendingPoolAddressesProvider} from "./ILendingPoolAddressesProvider.sol";
 import {DataTypes} from "../protocol/libraries/types/DataTypes.sol";
-import {IRescue} from "./../../interfaces/IRescue.sol";
+import {IRescue} from "../../../../interfaces/IRescue.sol";
 
 interface ILendingPool is IRescue {
     /**
      * @dev Emitted on deposit()
@@ -435,7 +435,5 @@
 
     function setPause(bool val) external;
 
     function paused() external view returns (bool);
-
-    function swapToVariable(address asset, address user) external;
 }
