===================================================================
--- src/contracts/protocol/lendingpool/LendingPoolStorage.sol
+++ contracts/v2AvaPool/LendingPool/contracts/protocol/lendingpool/LendingPoolStorage.sol
@@ -22,5 +22,11 @@
 
     uint256 internal _reservesCount;
 
     bool internal _paused;
+
+    uint256 internal _maxStableRateBorrowSizePercent;
+
+    uint256 internal _flashLoanPremiumTotal;
+
+    uint256 internal _maxNumberOfReserves;
 }
