Index: settings
===================================================================
--- settings
+++ settings
@@ -1,25 +1,9 @@
 {
-  "remappings": [
-    "@aave/core-v3/=lib/aave-helpers/lib/aave-address-book/lib/aave-v3-core/",
-    "@aave/periphery-v3/=lib/aave-helpers/lib/aave-address-book/lib/aave-v3-periphery/",
-    "aave-address-book/=lib/aave-helpers/lib/aave-address-book/src/",
-    "aave-helpers/=lib/aave-helpers/src/",
-    "aave-v3-core/=lib/aave-helpers/lib/aave-address-book/lib/aave-v3-core/",
-    "aave-v3-periphery/=lib/aave-helpers/lib/aave-address-book/lib/aave-v3-periphery/",
-    "ds-test/=lib/aave-helpers/lib/forge-std/lib/ds-test/src/",
-    "forge-std/=lib/aave-helpers/lib/forge-std/src/",
-    "governance-crosschain-bridges/=lib/aave-helpers/lib/governance-crosschain-bridges/",
-    "solidity-utils/=lib/aave-helpers/lib/solidity-utils/src/"
-  ],
   "optimizer": {
     "enabled": true,
     "runs": 200
   },
-  "metadata": {
-    "useLiteralContent": false,
-    "bytecodeHash": "none"
-  },
   "outputSelection": {
     "*": {
       "*": [
         "evm.bytecode",
@@ -30,14 +14,13 @@
         "abi"
       ]
     }
   },
-  "evmVersion": "istanbul",
   "libraries": {
-    "src/contracts/protocol/libraries/logic/ReserveLogic.sol": {
-      "ReserveLogic": "0xe58575ba47a348e3c2f9b7ec3eccfbb189ccc6ec"
+    "contracts/v2AvaPool/LendingPool/contracts/protocol/libraries/logic/ReserveLogic.sol": {
+      "ReserveLogic": "0xb5a1fe36dcf5ba149cb8d90a03f4709141ee5442"
     },
-    "src/contracts/protocol/libraries/logic/ValidationLogic.sol": {
-      "ValidationLogic": "0x2fba77a9dD1CcA8c3B54Ce20191057EAa08402D6"
+    "contracts/v2AvaPool/LendingPool/contracts/protocol/libraries/logic/ValidationLogic.sol": {
+      "ValidationLogic": "0xabcaf379c31626190f275f9a5ba5d6ebf19f9903"
     }
   }
 }
\ No newline at end of file
